stdlib/gui/gui.lua

```diff
@@ -33,6 +33,8 @@ function Gui.Event.register(event, gui_element_pattern, handler)
     Gui.Event._registry[event][gui_element_pattern] = handler
 
     -- Use custom Gui event dispatcher to pass off the event to the correct sub-handler
+    -- Remove our existing handler if it exists to prevent multiple event handlers.
+    Event.remove(event, Gui.Event.dispatch)
```

Comment:
I'd track a boolean here, initialized to false (`Gui._dispatch_registered` ? Feel free to use a better name) and only register on the first time, when it is false, then set it to true. Otherwise, we are doing a bunch of work to updating the event registry for no reason.